initfiles/bin/init_esp

```diff
@@ -33,12 +33,15 @@ SNMPID=$$
 
 killed()
 {
+    local __processID=$(cat ${PID_NAME})
     kill_process ${PID_NAME} esp 15 ${SENTINEL}
     if [[ $? -eq 1 ]]; then
         log "could not kill $component"
     else
         log "$component Stopped"
         unlock /var/lock/HPCCSystems/$component/${component}.lock
+        log "removing artifacts for process ID ${__processID} from /tmp"
+        find /tmp -name "*${__processID}" -exec rm -rf {} \;
```

Comment:
This cleans up the java_pid files as well as the java files that are in /tmp/hsperfdata_hpcc/${pid}